app/Models/Asset.php

```diff
@@ -907,6 +907,23 @@ public function requireAcceptance()
 
     }
 
+
+    /**
+     * Determine whether this asset's next audit date is before the last audit date
+     *
+     * @return bool
+     * @since [v6.4.1]
+     * @author [A. Gianotto] [<snipe@snipe.net>]
+     * */
+    public function checkInvalidNextAuditDate()
+    {
+        if (($this->last_audit_date) && ($this->next_audit_date) && ($this->last_audit_date > $this->next_audit_date)) {
+            return true;
+        }
+        return false;
```

Comment:
This is comparing a carbon instance and a string (from the . I'm not positive it'll be consistent but we should probably wrap `next_audit_date` in a `Carbon::parse` and then do the comparison.

(hopefully not necessary if my comment below works.

Reply:
Since we're still casting `next_audit_date` as a date time in the `protected $casts` section of the model, from my tests at least, it's always going to start off as a carbon object, so I think it's safe here - happy to fix if that turns out not to be true tho.

Reply:
I just read your comment above and you mention the serialization parts I mentioned here. Forgive the repetition.